packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterFetchCognitoAuthSessionResult.kt

```diff
@@ -37,10 +37,25 @@ data class FlutterFetchCognitoAuthSessionResult(private val raw: AWSCognitoAuthS
       "identityId" to this.identityId,
       "userSub" to this.userSub,
       "credentials" to this.credentials.serializeToMap(),
-      "tokens" to this.tokens.serializeToMap()
+      "tokens" to serializeTokens(this.tokens)
     )
   }
 
+  //parse userpool tokens
+  fun serializeTokens(res: AuthSessionResult<AWSCognitoUserPoolTokens>?): Map<String, Any>? {
+    var map = res.serializeToMap();
+    return if (map != null && map.containsKey("value")) {
+      var values = map["value"] as Map<String, Any>
+      if (!values.containsKey("error")) {
+        return values
+      } else {
+        return null
```

Comment:
do we need a `return` keyword on these since you are returning the whole `if`

Reply:
Looks like I just needed to lift both returns out of their enclosing 'ifs'.